samza-sql/src/main/java/org/apache/samza/sql/translator/MessageStreamCollector.java

```diff
@@ -150,65 +153,90 @@ public void close() {
     }
   }
 
+  /** {@inheritDoc} */
   @Override
-  public <OM> MessageStream<OM> flatMap(FlatMapFunction<? super SamzaSqlRelMessage, ? extends OM> flatMapFn) {
+  public <OM> MessageStream<OM> flatMap(String desc, FlatMapFunction<? super SamzaSqlRelMessage, ? extends OM> flatMapFn) {
     return null;
   }
 
+  /** {@inheritDoc} */
   @Override
   public <OM> MessageStream<OM> flatMapAsync(
+      String desc,
       AsyncFlatMapFunction<? super SamzaSqlRelMessage, ? extends OM> asyncFlatMapFn) {
     return null;
   }
 
+  /** {@inheritDoc} */
   @Override
-  public void sink(SinkFunction<? super SamzaSqlRelMessage> sinkFn) {
+  public void sink(String desc, SinkFunction<? super SamzaSqlRelMessage> sinkFn) {
     throw new IllegalStateException("Not valid state");
   }
 
+  /** {@inheritDoc} */
   @Override
-  public MessageStream<SamzaSqlRelMessage> sendTo(OutputStream<SamzaSqlRelMessage> outputStream) {
+  public MessageStream<SamzaSqlRelMessage> sendTo(String desc, OutputStream<SamzaSqlRelMessage> outputStream) {
     throw new IllegalStateException("Not valid state");
   }
 
+  /** {@inheritDoc} */
   @Override
-  public <K, WV> MessageStream<WindowPane<K, WV>> window(Window<SamzaSqlRelMessage, K, WV> window, String id) {
+  public <K, WV> MessageStream<WindowPane<K, WV>> window(String desc, Window<SamzaSqlRelMessage, K, WV> window, String id) {
```

Comment:
Would it make sense to have "desc" as an actual POJO so that it can be versioned in case one needs to evolve it in the future, e.g., 
pass in additional information about the operators between the layers, without that information necessarily having to be in the metric name. 

Having it as a POJO would also help get around future "backward compat" issues, versioning, etc.